### PR TITLE
T-02: Add the possibility to create new item

### PR DESCRIPTION
**Pull request by No1Nero**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -25,6 +25,14 @@
...
     -->
     <title>Todo Editor</title>
+    <style>
+      body{
+        margin: 0;
+        padding: 0;
+        margin-right: calc(-1 * (100vw - 100%));
+        overflow-x: hidden;
+      }
+    </style>
```
Comment (Yuliia143) [inline public/index.html]: Should this be here ?

File: src/components/TodoList/TodoList.module.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.list{
+    display: flex;
+    flex-wrap: wrap;
+    margin-left: 0;
+    padding-left: 0;
+}
+.item_wrapper{
+    list-style-type: none;
```
Comment (Yuliia143) [inline src/components/TodoList/TodoList.module.css]: Please, add normalize or reset file
Normalize - makes all browsers render elements the same way, and then you can write styles for list as list-style-type: none; inside some general CSS file
Reset - it resets default styles

Better to have normalize + write such styles in general CSS file

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,54 @@
...
+  const dispatch = useDispatch();
+
+  const addTodo = (todo: ITodo) => {
+    dispatch({type: TodoActionTypes.CREATE_TODO, payload: todo});
+    setCreateInputText('');
+  };
+
+  const handleInputText = (e: React.FormEvent<HTMLInputElement>) => {
+    const formattedInputText = e.currentTarget.value.replace(/[{}+/@#$%^&*|;:<>=_]/g, '');
+    if (formattedInputText.length > 70) {
```
Comment (Yuliia143) [inline src/App.tsx]: It's better to create  const for the magic number

File: src/App.tsx [hunk 4]
```diff
@@ -1,7 +1,54 @@
...
+    var hours = date.getHours();
+    var minutes = date.getMinutes();
+
+    day = (day < 10) ? '0' + day : day;
+    month = (month < 10) ? '0' + month : month;
+    hours = (hours < 10) ? '0' + hours : hours;
+    minutes = (minutes < 10) ? '0' + minutes : minutes;
+
+    return day + '.' + month + '.' + year + ' ' + hours + ':' + minutes;
+  };
```
Comment (Yuliia143) [inline src/App.tsx]: Shouldn't be inside the component - create utils/helpers folder for that

File: src/App.tsx [hunk 5]
```diff
@@ -1,7 +1,54 @@
...
+
   return (
-    <div>
-      App works well
+    <div style={isShownModal ? {overflowY: 'hidden', overflowX: 'hidden', height: '100vh'} : {overflowY: 'visible'}}>
```
Comment (Yuliia143) [inline src/App.tsx]: It's better not to use inline styles

File: src/components/CreateInput/CreateInput.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        if (createInputText.trim()) {
+            const obj = {
+                id: Math.random().toString(36).substr(2, 9),
+                title: createInputText,
+                status: false,
+                creationDate: onFormatDate(new Date()),
+                expirationDate: onFormatDate(new Date(Date.now() + 1000*60*60*24)),
```
Comment (Yuliia143) [inline src/components/CreateInput/CreateInput.tsx]: Magic numbers x2

File: src/components/CreateInput/CreateInput.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    onFormatDate: any
+};
+
+export default function CreateInput({createInputText, onHandleInputText, onAddTodo, onSetIsShownModal, onFormatDate}: CreateInputProps) {
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        if (createInputText.trim()) {
+            const obj = {
+                id: Math.random().toString(36).substr(2, 9),
```
Comment (Yuliia143) [inline src/components/CreateInput/CreateInput.tsx]: Magic numbers x3
Probably better to have a function that generates ID or to connect library that generate guids

File: src/components/CreateInput/CreateInput.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            onAddTodo(obj);
+        }
+    };
+
+    return (
+        <form className={styles.form} onSubmit={handleSubmit}>
+            <h2 className={styles.title}>Create Todo</h2>
+            <div className={styles.input_wrapper}>
+                <input type="text" className={styles.input} value={createInputText} onChange={onHandleInputText} placeholder="Enter title to create" />
+                <button onClick={()=> onSetIsShownModal(isShownModal => !isShownModal)} type="button" className={styles.button}><BsPlusLg size={40}/></button>
```
Comment (Yuliia143) [inline src/components/CreateInput/CreateInput.tsx]: Better to have function than write logic in JSX

File: src/components/Modal/Modal.tsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+            setExpires('');
+        } else {
+            setExpires(e.currentTarget.value);
+        }
+        
+    };
+
+    const publicTodo = () => {
+        const obj = {
+            id: Math.random().toString(36).substr(2, 9),
```
Comment (Yuliia143) [inline src/components/Modal/Modal.tsx]: Again - function or library for guids

File: src/components/Modal/Modal.tsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+            status: false,
+            creationDate: created,
+            expirationDate: onFormatDate(new Date(expires)),
+        };
+        onAddTodo(obj);
+        onSetIsShownModal(isShownModal => !isShownModal);
+    };
+
+    const cancelModal = () => {
+        onSetIsShownModal(isShownModal => !isShownModal);
```
Comment (Yuliia143) [inline src/components/Modal/Modal.tsx]: Something that repeats 2 or more times should be a function

File: src/components/Modal/Modal.tsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        onSetCreateInputText('');
+    };
+
+    return (
+        <div className={styles.modal_window}>
+            <div onClick={()=> onSetIsShownModal(isShownModal => !isShownModal)} className={styles.modal_blur}></div>
```
Comment (Yuliia143) [inline src/components/Modal/Modal.tsx]: Logic in JSX

File: src/components/Modal/Modal.tsx [hunk 12]
```diff
@@ -0,0 +1,73 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+import styles from './Modal.module.css';
+
+interface ModalProps {
+    onSetIsShownModal: Dispatch<SetStateAction<boolean>>,
+    createInputText: string,
+    onHandleInputText: any,
+    onAddTodo: any,
+    onFormatDate: any,
```
Comment (Yuliia143) [inline src/components/Modal/Modal.tsx]: Why any ?

File: src/components/CreateInput/CreateInput.tsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import { Dispatch, SetStateAction } from "react";
+import styles from './CreateInput.module.css';
+import {BsPlusLg} from 'react-icons/bs';
+
+interface CreateInputProps {
+    createInputText: string,
+    onHandleInputText: any,
+    onAddTodo: any,
+    onSetIsShownModal: Dispatch<SetStateAction<boolean>>,
+    onFormatDate: any
```
Comment (Yuliia143) [inline src/components/CreateInput/CreateInput.tsx]: Why any?

File: src/components/CreateInput/CreateInput.tsx [hunk 14]
```diff
@@ -0,0 +1,38 @@
+import { Dispatch, SetStateAction } from "react";
+import styles from './CreateInput.module.css';
+import {BsPlusLg} from 'react-icons/bs';
```
Comment (Yuliia143) [inline src/components/CreateInput/CreateInput.tsx]: Follow one code style
different indents {BsPlusLg} and { Dispatch, SetStateAction }

File: src/components/Modal/Modal.tsx [hunk 15]
```diff
@@ -1,18 +1,21 @@
...
+    onSetIsShownModal: Dispatch<SetStateAction<boolean>>,
     onSetCreateInputText: Dispatch<SetStateAction<string>>
 };
 
-export default function Modal({onSetIsShownModal, createInputText, onHandleInputText, onAddTodo, onFormatDate, onSetCreateInputText}: ModalProps) {
-    const [created] = useState<string>(onFormatDate(new Date()));
+export default function Modal({createInputText, onHandleInputText, onAddTodo, onSetIsShownModal, onSetCreateInputText}: ModalProps) {
+    const [created] = useState<string>(formatDate(new Date()));
     const [expires, setExpires] = useState<string>('');
+    const todayMinDate = new Date().toISOString().slice(0, 16);
```
Comment (Yuliia143) [inline src/components/Modal/Modal.tsx]: Magic number

File: src/index.css [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+    margin: 0;
+    padding: 0;
+    margin-right: calc(-1 * (100vw - 100%));
+    overflow-x: hidden;
+}
+ul{
+    list-style-type: none;
+    margin-left: 0;
+    padding-left: 0;
+}
```
Comment (Yuliia143) [inline src/index.css]: Empty line

File: src/utils/formatDate.ts [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+    var day = date.getDate();
+    var month = date.getMonth() + 1;
+    var year = date.getFullYear();
+    var hours = date.getHours();
+    var minutes = date.getMinutes();
+
+    day = (day < 10) ? '0' + day : day;
+    month = (month < 10) ? '0' + month : month;
+    hours = (hours < 10) ? '0' + hours : hours;
+    minutes = (minutes < 10) ? '0' + minutes : minutes;
```
Comment (Yuliia143) [inline src/utils/formatDate.ts]: Magic number

File: src/components/Modal/Modal.tsx (hunk 15 above)
Comment (No1Nero) [reply src/components/Modal/Modal.tsx]: This is a variable to disabling past dates in the input. Today we discussed magic numbers and tou said that they should be variables and not just in the code. Please, tell me how to do it

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**